File: 1711-find-valid-matrix-given-row-and-column-sums/find-valid-matrix-given-row-and-column-sums.java
class Solution {
    public int[][] restoreMatrix(int[] rowSum, int[] colSum) {
        int x = rowSum.length;
        int y = colSum.length;
        int[][] res = new int[x][y];

        for (int i = 0; i < x; i++) {
            for (int j = 0; j < y; j++) {
                int val = Math.min(rowSum[i], colSum[j]);
                res[i][j] = val;
                rowSum[i] -= val;
                colSum[j] -= val;
            }
        }

        return res;
    }
}

File: 1711-find-valid-matrix-given-row-and-column-sums/find-valid-matrix-given-row-and-column-sums.py
class Solution(object):
    def restoreMatrix(self, rowSum, colSum):
        x, y = len(rowSum), len(colSum)
        res = [[0] * y for _ in range(x)]

        for i in range(x):
            for j in range(y):
                val = min(rowSum[i], colSum[j])
                res[i][j] = val
                rowSum[i] -= val
                colSum[j] -= val
        
        return res
